app/services/verify_link_service.rb

```diff
@@ -5,9 +5,11 @@ def call(field)
     @link_back = ActivityPub::TagManager.instance.url_for(field.account)
     @url       = field.value_for_verification
 
-    perform_request!
+    unless link_redirects_back?(@url)
```

Comment:
Since in most cases the link would not redirect back to the profile itself, I think it's best to avoid an extra request. We can probably rely on `res.uri` in `perform_request!` to find out whether the request redirected to the account directly.